src/fielddef.ts

```diff
@@ -285,22 +286,9 @@ export function normalize(channelDef: ChannelDef<string>, channel: Channel) {
     }
 
     // Normalize bin
-    if (fieldDef.bin) {
-      const bin = fieldDef.bin;
-      if (isBoolean(bin)) {
-        fieldDef = {
-          ...fieldDef,
-          bin: {maxbins: autoMaxBins(channel)}
-        };
-      } else if (!bin.maxbins && !bin.step) {
-        fieldDef = {
-          ...fieldDef,
-          bin: {
-            ...bin,
-            maxbins: autoMaxBins(channel)
-          }
-        };
-      }
+    const bin = normalizeBin(fieldDef.bin, channel);
```

Comment:
Why setting fieldDef.bin will cause side effect? I see what we had before is actually overwriting fieldDef.bin, isn't it?